config/crds/patches/elasticsearch-patches.yaml

```diff
@@ -1,3 +1,13 @@
+# Move validation of the v1 version to the top-level, and remove validation from the v1beta1 version.
+# This is the behaviour intended from using controller-tools crd:trivialVersions=true flag,
+# which currently does not work as expected.
+# See https://github.com/kubernetes-sigs/controller-tools/issues/349.
+# TODO: remove once fixed in controller-tools
+- op: move
+  from: /spec/versions/0/schema
+  path: /spec/validation
+- op: remove
+  path: /spec/versions/1/schema
```

Comment:
Since ES, Kibana & APM share this same exact block I initially moved it to `patches/all-patches.yaml` but it didn't work well.
Is it because patches in the same file are applied sequentially vs. concurrently in different files, or something along those lines?

Reply:
Can you clarify the issue with it in `all-patches.yaml`? I was curious and so I checked it out and moved it there. It looks like it generates the same output for me on controller-tools v0.2.1.

Reply:
Since this is kustomize running after initial CRD generation is done it should not have anything to do with controller-tools version?
I updated kustomize to the latest version (3.5.4) and gave it another try.
When moving
```
# Move validation of the v1 version to the top-level, and remove validation from the v1beta1 version.
# This is the behaviour intended from using controller-tools crd:trivialVersions=true flag,
# which currently does not work as expected.
# See https://github.com/kubernetes-sigs/controller-tools/issues/349.
# TODO: remove once fixed in controller-tools
- op: move
  from: /spec/versions/0/schema
  path: /spec/validation
- op: remove
  path: /spec/versions/1/schema
# Remove validation.openAPIV3Schema.type that causes failures on k8s 1.11.
# This should have been fixed with https://github.com/kubernetes-sigs/controller-tools/pull/72, but it looks like
# this commit has been lost in history. See https://github.com/kubernetes-sigs/controller-tools/issues/296.
# TODO: remove once fixed in controller-tools
- op: remove
  path: /spec/validation/openAPIV3Schema/type
```
from `apm-kibana-patches.yaml` and `elasticsearch-patches.yaml` to `all-patches.yaml`, then run `make generate` I get the following error:

```
failed to apply json patch '
# Using `kubectl apply` stores the complete CRD file as an annotation,
# which may be too big for the annotations size limit.
# One way to mitigate this problem is to remove the (huge) podTemplate properties from the CRD.
# It also avoids the problem of having any k8s-version specific field in the Pod schema,
# that would maybe not match the user's k8s version.
- op: remove
  path: /spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties
': jsonpatch remove operation does not apply: doc is missing path: "/spec/validation/openAPIV3Schema/properties/spec/properties/podTemplate/properties"
make: *** [generate-crds] Error 1
```

I think that's because the patches in `elasticsearch-patches.yaml` rely on the patches in `all-patches.yaml`. Switching the order of items in `patchesJson6902` in `kustomization.yaml` does not seem to make any difference.

Reply:
As discussed out of band I was only trying to move
```
- op: remove
  path: /spec/versions/1/schema
```
and it's not worth moving just that command to the `all-patches.yaml`, as we would also need to copy and paste the comment. I ran into the same error you did if I try to move the
```
- op: move
  from: /spec/versions/0/schema
  path: /spec/validation
```
stanza